smarts/core/sensors.py

```diff
@@ -318,6 +319,40 @@ def observe(sim, agent_id, sensor_state, vehicle):
     def step(sim, sensor_state):
         return sensor_state.step()
 
+    @classmethod
+    def _agents_alive_done_check(
+        cls, agent_manager, agents_alive: AgentsAliveDoneCriteria
+    ):
+        if not agents_alive:
+            return False
+
+        done = False
+        if agents_alive.total_ego_agents_alive:
+            done = (
+                len(agent_manager.ego_agent_ids) < agents_alive.total_ego_agents_alive
+            )
+        if not done and agents_alive.total_social_agents_alive:
+            done = (
+                len(agent_manager.social_agent_ids)
+                < agents_alive.total_social_agents_alive
+            )
+        if not done and agents_alive.agent_lists_alive:
+            for agent_list, min_alive in agents_alive.agent_lists_alive:
+                assert isinstance(
+                    agent_list, (List, Set, Tuple)
+                ), "Please specify a list of agent ids to watch"
+                assert isinstance(
+                    min_alive, int
+                ), "Please specify an int for minimum number of alive agents in the list"
+                agents_alive = [
+                    1 if id in agent_manager.agent_ids else 0 for id in agent_list
+                ]
+                done = agents_alive.count(1) < min_alive
+                if done:
+                    break
```

Comment:
...and could just `return True` here too.  (with the default "fall through" case below being `return False`.)